modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -597,6 +606,12 @@ public File snapshotTmpDir() {
                 "prior to snapshot operation start: " + leftNodes));
         }
 
+        if (!cctx.localNode().isClient() && cctx.kernalContext().encryption().isMasterKeyChangeInProgress()
```

Comment:
In the construction `A && B || C` the C operation will be invoked despite of whether the A is true of false. So it may lead to NPE, as cctx is `null` for client nodes.

Reply:
If cctx is null for client, we fail at start: `if (cctx.kernalContext().clientNode()`. Also snapshots is disables for client nodes because clisents do not have data and have nothing to snapshot.

Reply:
I don't think so. We do not affect snapshot-on-client behavior in this ticket. It has been working well till now. Shapshots-on-client hasn't raised NPE before the ticket. I believe, because this DistributedProcess doesn't work on client regardless encrypted or not.

Reply:
Yes, you're right. `GridCacheContext` is null for client node in `GridQueryProcessor` only.

So, what about to fix boolean logic then - `A && (B || C)`?

Reply:
Fixed

Reply:
Fixed